### ISSUE TITLE
`improper_ctypes` lint should cover "C" abi function pointer types.

### ISSUE THREAD

**Issue by crlf0710** [rust-lang/rust#95683]
I tried this code:

```rust
pub type F = extern "C" fn(&[u8]);
```

I expected to see this happen: `improper_ctypes` should trigger for this definition.

Instead, this happened: It is not linted.
